packages/grafana-alerting/scripts/codegen.ts

```diff
@@ -1,21 +1,48 @@
 /**
- * This script will generate TypeScript type definitions and a RTKQ client for the alerting k8s APIs.
- * It downloads the OpenAPI schema from a running Grafana instance and generates the types.
+ * This script will generate TypeScript type definitions and a RTKQ clients for the alerting k8s APIs.
  *
  * Run `yarn run codegen` from the "grafana-alerting" package to invoke this script.
+ *
+ * API clients will be placed in "src/grafana/api/<version>/api.gen.ts"
  */
-import { type ConfigFile } from '@rtk-query/codegen-openapi';
+import type { ConfigFile } from '@rtk-query/codegen-openapi';
 import { resolve } from 'node:path';
 
-// these snapshots are generated by running "go test pkg/tests/apis/openapi_test.go", see the README in the "openapi_snapshots" directory
-const OPENAPI_SCHEMA_LOCATION = resolve(
-  '../../../pkg/tests/apis/openapi_snapshots/notifications.alerting.grafana.app-v0alpha1.json'
-);
+// ℹ️ append versions here to generate additional API clients
+const VERSIONS = ['v0alpha1'] as const;
+
+type OutputFile = Omit<ConfigFile, 'outputFile'>;
+type OutputFiles = Record<string, OutputFile>;
+
+const outputFiles = VERSIONS.reduce<OutputFiles>((acc, version) => {
+  // we append the version here so we export versioned API clients from this package without having to re-export with an alias
+  const exportName = `alertingAPI${version}`;
+
+  // ℹ️ these snapshots are generated by running "go test pkg/tests/apis/openapi_test.go" and "sripts/process-specs.sh",
+  // see the README in the "openapi_snapshots" directory
+  const schemaFile = resolve(`../../../data/openapi/notifications.alerting.grafana.app-${version}.json`);
```

Comment:
Are these `resolve` wrappers necessary? Seems to work without them in the core.

Reply:
No I don't think they're technically required when invoked through yarn (since that forces the same cwd all the time).